saleor/wishlist/views.py

```diff
@@ -0,0 +1,92 @@
+import json
+
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.http import JsonResponse
+from django.shortcuts import get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.views.decorators.http import require_POST
+
+from ..core.utils import get_paginator_items
+from ..product.models import Product, ProductVariant
+from . import utils
+from .models import Wishlist, WishlistItem
+
+
+@login_required
+def user_wishlist(request):
+    """
+    :type request: HttpRequest
+    :rtype: HttpResponse
+    """
+    wishlist, created = Wishlist.objects.get_or_create(user=request.user)
+    return redirect('wishlist:public-wishlist', token=wishlist.token)
+
+
+def public_wishlist(request, token):
+    """
+    :type request: HttpRequest
+    :type token: str
+    :rtype: HttpResponse
+    """
+    wishlist = get_object_or_404(Wishlist, token=token)
+    if not wishlist.public and request.user != wishlist.user:
+        return TemplateResponse(request, 'wishlist/access_denied.html', {})
```

Comment:
`get_object_or_404` takes model or queryset, so you can construct a queryset that filters only public wishlists, so when user is not allowed to see a wishlist, he will see standard 404 page.